openverse_catalog/dags/common/requester.py

```diff
@@ -50,23 +49,24 @@ def __init__(self, delay=0, headers=None):
         self._last_request = 0
         self.session = requests.Session()
 
-    def get(self, url, params=None, **kwargs):
+    def _make_request(self, method, url, **kwargs):
```

Comment:
```suggestion
    def _make_request(self, method: Literal['get', 'head'], url: str, **kwargs):
```

Reply:
👋 Hi @obulat - thanks for the suggestion. Could you help me understand this literal type constraint better? I was under the impression that the calls to `_make_request` were passing methods from the Requests module and not string values, but I'm probably missing something.

To experiment I changed the `Literal` params to something nonsensical (instead of `get` and `head`) and `just lint` and `just test` passed when I was expecting them to fail.

Also, if you're suggesting that we make the Requests module calls only within `_make_request` based on the string passed then that makes sense, but I wanted to get confirmation. Thank you!

Reply:
Gah, that's right! This is actually of type `callable`, rather than `Literal`. I think the way it is presently is fine, but it might be worth adding the generic `callable` type annotation (trying to match the call signatures of both `.head` and `.get` will be a _head_-ache 😄)

Reply:
Thank you for stepping in, @AetherUnbound, and sorry I couldn't reply sooner, @twstokes.